bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/service/SmartThingsApiService.java

```diff
@@ -470,15 +474,14 @@ public String getTvChannelName() {
         }
     }
 
-    public Number parseTVChannel(@Nullable String channel) {
+    public static Number parseTVChannel(@Nullable String channel) {
```

Comment:
It fixes the NPE, but I'm not sure that it actually decodes the TV channel correctly. I tried this, and was getting -1 for the channel decoding, so I added some extra code, that then seemed to work correctly.

So I didn't want to just fix the NPE, I also wanted it to work as expected, unfortunately I can't test it myself, as I don't have DTV.

Reply:
And to what code are you referring?

The only related change i see is that the caller `getTvChannel()` has the `.filter(i -> !i.isBlank()` removed, and that is now added to the `parseTVChannel`.

Reply:
I added:
```
        class TvChannel {
            Values tvChannel;
            Values tvChannelName;

            public String getChannelNum() {
                return Optional.ofNullable(tvChannel).map(a -> a.value).orElse("");
            }
        }
```
`getChannelNum()` because the class is called 'tvChannel' and the value is also in a class called 'tvChannel' (because this is how the json is structured), and it was always returning null when parsing the json received from Smartthings.

So, it wasn't the parsing that was the problem, it was the decoding of the json, and extracting the values. I continually got null, until I implemented the above.

Reply:
Please extract these changes related to the NPE and these `getChannelNum()` to a seperate PR.

Reply:
I extracted the parts for you in a seperate PR that can be used to backport.  https://github.com/openhab/openhab-addons/pull/17454
Once merged, you might need to fix any conflicts.